src/commands/open.ts

```diff
@@ -0,0 +1,101 @@
+}
+
+const LINKS: Link[] = [
+  { name: "Analytics", arg: "analytics", consoleUrl: "/analytics" },
+  { name: "Authentication: Providers", arg: "auth", consoleUrl: "/authentication/providers" },
+  { name: "Authentication: Users", arg: "auth:users", consoleUrl: "/authentication/users" },
+  { name: "Crash Reporting", arg: "crash", consoleUrl: "/monitoring" },
+  { name: "Database: Data", arg: "database", consoleUrl: "/database/data" },
+  { name: "Database: Rules", arg: "database:rules", consoleUrl: "/database/rules" },
+  { name: "Docs", arg: "docs", url: "https://firebase.google.com/docs" },
+  { name: "Dynamic Links", arg: "links", consoleUrl: "/durablelinks" },
+  { name: "Hosting: Deployed Site", arg: "hosting:site" },
+  { name: "Hosting", arg: "hosting", consoleUrl: "/hosting/main" },
+  { name: "Notifications", arg: "notifications", consoleUrl: "/notification" },
+  { name: "Project Dashboard", arg: "dashboard", consoleUrl: "/overview" },
+  { name: "Project Settings", arg: "settings", consoleUrl: "/settings/general" },
+  { name: "Remote Config: Conditions", arg: "config:conditions", consoleUrl: "/config/conditions" },
+  { name: "Remote Config", arg: "config", consoleUrl: "/config" },
+  { name: "Storage: Files", arg: "storage", consoleUrl: "/storage/files" },
+  { name: "Storage: Rules", arg: "storage:rules", consoleUrl: "/storage/rules" },
+  { name: "Test Lab", arg: "testlab", consoleUrl: "/testlab/histories/" },
+];
+
+const CHOICES = _.map(LINKS, "name");
+
+export default new Command("open [link]")
+  .description("quickly open a browser to relevant project resources")
+  .before(requirePermissions)
+  .before(requireInstance)
+  .action(
+    async (linkName: string, options: any): Promise<void> => {
+      let link = _.find(LINKS, { arg: linkName });
+      if (linkName && !link) {
+        throw new FirebaseError(
+          "Unrecognized link name. Valid links are:\n\n" + _.map(LINKS, "arg").join("\n")
+        );
+      }
+
+      if (!link) {
+        const name = await promptOnce({
+          type: "list",
+          message: "What link would you like to open?",
+          choices: CHOICES,
+        });
+        link = _.find(LINKS, { name });
+      }
+      if (!link) {
+        throw new FirebaseError(
+          "Unrecognized link name. Valid links are:\n\n" + _.map(LINKS, "arg").join("\n")
+        );
+      }
+
+      let url;
+      if (link.consoleUrl) {
+        url = utils.consoleUrl(options.project, link.consoleUrl);
+      } else if (link.url) {
+        url = link.url;
+      } else if (link.arg === "hosting:site") {
+        url = utils.addSubdomain(api.hostingOrigin, options.instance);
+      } else if (link.arg === "functions") {
+        url = "https://console.firebase.google.com/project/" + options.project + "/functions/list";
+      } else if (link.arg === "functions:log") {
+        url =
+          "https://console.developers.google.com/logs/viewer?resource=cloudfunctions.googleapis.com&project=" +
```

Comment:
Same as above, move to const at the top of the file as `GCP_CONSOLE = ...`

Reply:
I weigh these in my head: moving a string to a constant elsewhere in a file vs. keeping context where it's necessary. In this case, keeping the context well out-weighs moving part of this to a constant. It's only in one spot and it's important to realize this goes very much elsewhere.

That said, isn't this a really, really outdated URL (should it be `console.cloud.google.com`)? And why shouldn't it be `utils.consoleUrl(options.project, "/functions/logs")`?

Reply:
Yeah, I agree if it's not used anywhere else that it's fine to keep the URL here. Interesting point, this was probably going to the google cloud console because we didn't have logs implemented in firebase console? We should change that to open the Firebase console :)

Reply:
Okay. I'm fine with changing that, but I think that should be in a different CL since that's changing some behavior (I'd like to keep this having the same behavior... plus that little fix)

Reply:
sure that sounds fine